### ISSUE TITLE
Navigator not showing chart for line chart

### ISSUE THREAD

**Issue by mynameiswhat1** [highcharts/highcharts#7686]
#### Expected behaviour
Navigator should show the chart line.

#### Actual behaviour
Navigator does not show the chart line, as the following picture ilustrates:

<a href="https://ibb.co/jXuiRm"><img src="https://preview.ibb.co/c4NsK6/nav.jpg" alt="nav" border="0"></a>

#### Live demo with steps to reproduce
After the following demo is run, navigator is empty:
http://jsfiddle.net/rx0mnhpw/

#### Product version
Highcharts 5.0.14

#### Affected browser(s)
Version 63.0.3239.84 (Official Build) (64-bit)

**Comment by pawelfus** [comment]
Hi @mynameiswhat1 

This is not a bug - this is caused by `dataGrouping` which by default is enabled in navigator (and strongly recommended). With your current config (`gapSize`=1minute) and default `dataGrouping` we will receive points with ~15minutes (depending on a chart size can be 5minutes, 10minutes, 15minutes etc.) gaps between. You can enable dataGrouping in your demo and see the same result for the main chart, see: http://jsfiddle.net/BlackLabel/rx0mnhpw/2/

**Comment by mynameiswhat1** [comment]
Hello @pawelfus ,
Sorry, but I do not understand how can this not be a bug. I have a data set with a sampling rate of 1 minute. I wish to see a gap in the chart if there is an interval greater than 1 minute for which I do not receive data, hence the gapSize=60000. However, if I set highcharts as such I cannot see the data in navigator. For me, this does not make any sense, and for this reason is a bug.
Thank you.

**Comment by mynameiswhat1** [comment]
Hello @pawelfus ,
By enabling dataGrouping in the navigator we get the same results:

[http://jsfiddle.net/53oopd9q/](http://jsfiddle.net/53oopd9q/)

I believe that this is a bug. 
Thank you.

**Comment by pawelfus** [comment]
Hi again @mynameiswhat1 

Let me rephrase:

Navigator has default setting as  `navigator.series.dataGrouping.enabled = true`, which causes the issue. In your case, dataGrouping for navigator groups points into packages 5/10/15minutes (depending on a screen size) and renders a point every 5/10/15 minutes. 
It's not a bug, because your setting (`gapSize=1minute`) with default grouped points (e.g. 5minutes distance) shouldn't draw a line between points (5minutes distance between points > 1minute gapSize). You can see the line if navigator has **disabled** dataGrouping: http://jsfiddle.net/BlackLabel/rx0mnhpw/3/

@TorsteinHonsi tl;dr:
`gapSize` is a great option, but it should be possible to set it per `dataGrouping` unit (e.g. daily grouping = daily `gapSize`, 2weeks grouping = 2weeks `gapSize` etc.). In the case above we have data based on 1minute and we want to show gaps when point is missing. However, when dataGrouping turns on and groups points into 5minute packages, then every point is rendered as a detached one.

**Comment by mynameiswhat1** [comment]
Hello @pawelfus ,
Yes, I misunderstood your statement about the navigator datagroupping. With it disabled, the line is shown: [http://jsfiddle.net/fts2c5b1/](http://jsfiddle.net/fts2c5b1/)

Regarding the statement:

> Navigator has default setting as navigator.series.dataGrouping.enabled = true, which causes the issue. In your case, dataGrouping for navigator groups points into packages 5/10/15minutes (depending on a screen size) and renders a point every 5/10/15 minutes.

Where can I find information that correlates the screen size with the 5/10/15 minutes packages?

> It's not a bug, because your setting (gapSize=1minute) with default grouped points (e.g. 5minutes distance) shouldn't draw a line between points (5minutes distance between points > 1minute gapSize). You can see the line if navigator has disabled dataGrouping: http://jsfiddle.net/BlackLabel/rx0mnhpw/3/

Maybe this is not a bug, from a code standpoint, however, I strongly believe that this is a huge issue, maybe from a design standpoint. In my point of view, Highcharts should be capable of showing the correct gaps in the data, as set by gapSize, being the data grouped or not ,otherwise, from a user perspective, it makes no sense. Also, this has been causing us an enormous array of problems, which have been discussed since April in https://github.com/highcharts/highcharts/issues/6645.
Thank you.

**Comment by pawelfus** [comment]
> Where can I find information that correlates the screen size with the 5/10/15 minutes packages?

In our API: [`navigator.series.dataGrouping`](https://api.highcharts.com/highstock/navigator.series.dataGrouping), units are explained here: [`navigator.series.dataGrouping.units`](https://api.highcharts.com/highstock/navigator.series.dataGrouping.units).

I agree with you, `gapSize` could be improved - we need some synergy between this option and `dataGrouping`. What do you think about my solution? Something like this:


1. If `gapSize` is set and points are **not** grouped, apply it directly (just like my demo with disabled `dataGrouping` - no issues there)
2. If `gapSize` is set **and** `dataGrouping` is used then: 

 - if `gapSize` is **higher** than grouped points granularity, use defined `gapSize`
 - if `gapSize` is **lower** than grouped points granularity, use that grouping 

In other words, for grouped points, `gapSize` will be dynamically calculated, like this: `currentGapSize = Math.max(userGapSize, dataGroupingDistance)`, e.g. `currentGapSize = Math.max(1minute, 15minutes)`

**Comment by mynameiswhat1** [comment]
> In our API: navigator.series.dataGrouping, units are explained here: navigator.series.dataGrouping.units.

Yes, but where is the correlation between this and screen size?


Your solution looks good. Will it be included in a future release of Highcharts? In the meantime, can you provide a workaround?

Thank you very much for your help @pawelfus .

**Comment by mynameiswhat1** [comment]
> Your solution looks good. Will it be included in a future release of Highcharts? In the meantime, can you provide a workaround?

Just came to my mind, what about when we don't have data grouping in the chart, but the navigator sets it by default, like in one of the fiddles above? There will be a currentGapZise for the chart and other for the navigator? This will be calculated internally by Highcharts, the user only has to set one gapSize?

Thank you.

**Comment by pawelfus** [comment]
Check all of the options in the dataGrouping - everything should be explained there. Correlation between screen size and dataGrouping is described in (and controlled by) [`navigator.series.dataGrouping.groupPixelWidth`](https://api.highcharts.com/highstock/navigator.series.dataGrouping.groupPixelWidth).

Let's assume this is the issue: http://jsfiddle.net/BlackLabel/3hromonu/ - I'm thinking about something like this: http://jsfiddle.net/BlackLabel/3hromonu/4/  (we can see it better without ordinal xAxis: http://jsfiddle.net/BlackLabel/3hromonu/5/ )

Workaround applied to your demo: http://jsfiddle.net/BlackLabel/2js9o87d/1/ - could you apply my snippet (below) and let me know if this works for you?

Snippet (load it after Highstock, like any other plugin):
```
(function(H) {
  H.wrap(H.Series.prototype, 'gappedPath', function(p) {
    var userGapSize = H.pick(
        this.userOptions.gapSize,
        this.options.gapSize
      ),
      groupingSize = this.currentDataGrouping ? this.currentDataGrouping.totalRange : 0;


    if (groupingSize && userGapSize && groupingSize > userGapSize) {
      this.options.gapSize = groupingSize;
    }

    return p.apply(this, Array.prototype.slice.call(arguments, 1));
  });
})(Highcharts);
```

**Comment by pawelfus** [comment]
Just realized, it's a duplicate of #7521 - which was decided we should improve docs with explanation how it works.

**Comment by mynameiswhat1** [comment]
Does this mean the enhancement will still be applied to future Highcharts releases or not?

**Comment by pawelfus** [comment]
I'm not sure, but if this will be a problem to more users we will improve logic in the core. For now please use my workaround.

**Comment by TorsteinHonsi** [comment]
@pawelfus I agree with your [comment above](https://github.com/highcharts/highcharts/issues/7686#issuecomment-358625697). If data grouping is greater than the `gapSize`, use the data grouping. I'm assigning this to you, feel free to delegate.
